### PR TITLE
NB-246: Allow users to provide JWT token

### PR DESCRIPTION
**Pull request by olgashev**

### CODE REVIEW

File: src/nebulr-auth/auth-guard.ts [hunk 1]
```diff
@@ -137,7 +179,7 @@ export class AuthGuard implements CanActivate {
   }
 
   private getGraphqlResource(graphqlCtx: GqlExecutionContext): string {
-    return `graphql/${graphqlCtx.getHandler().name}`;
+    return `graphql / ${graphqlCtx.getHandler().name} `;
```
Comment (orr3) [inline src/nebulr-auth/auth-guard.ts:182]: ??

File: src/nebulr-auth/auth-guard.ts [hunk 2]
```diff
@@ -45,54 +67,70 @@ export class AuthGuard implements CanActivate {
...
+
+        } else {
+          return false;
+        }
+      } catch (error) {
+        console.error(error);
+        throw error;
+      }
     } else {
-      const hasRequiredPlan = this.authGuardService.hasRequiredPlan(authResponse.user.tenant.plan, parsedRequest.resource);
```
Comment (orr3) [inline src/nebulr-auth/auth-guard.ts:92]: Where did the old way of using legacy authTokens go?

**Review by orr3 — CHANGES_REQUESTED** [review]

File: src/nebulr-auth/auth-guard.ts (hunk 1 above)
Comment (olgashev) [reply src/nebulr-auth/auth-guard.ts]: my fault. fixed

File: src/nebulr-auth/auth-guard.ts [hunk 3]
```diff
@@ -45,54 +67,70 @@ export class AuthGuard implements CanActivate {
...
-    }
-
-    // If granted, check tenant plan before returning
-    if (NebulrAuthService.isAnonymousUser(authResponse.user)) {
-      this._debugger.log(`canActivate no need to do plan restriction since this is a ANONYMOUS user`);
-      return true;
+    const acessRawToken = parsedRequest.request.get('Authorization') || parsedRequest.request.get('authorization');
+    if (acessRawToken && acessRawToken.startsWith('Bearer ')) {
+      const acessToken = acessRawToken.substring(7, acessRawToken.length);
+      const key = await this._jwksClient.getSigningKey();
```
[the remark this answers is not in the document]
Comment (olgashev) [reply src/nebulr-auth/auth-guard.ts]: changed with use of jose package

File: nblocks/config/dev.env [hunk 4]
```diff
@@ -0,0 +1,6 @@
+NBLOCKS_API_KEY="NBLOCKS_API_KEY"
+NBLOCKS_DEBUG=*
+NBLOCKS_FORCE_DEV=true
+
+NEBULR_PLATFORM_CORE_API_URL="http://account-api:3000"
+NBLOCKS_JWKS_URL="http://auth-api:3000/.well-known/jwks.json"
```
[the remark this answers is not in the document]
Comment (olgashev) [reply nblocks/config/dev.env]: removed

File: src/nebulr-auth/auth-guard.ts (hunk 2 above)
Comment (olgashev) [reply src/nebulr-auth/auth-guard.ts:92]: restored

**Pull request merged**